src/server/pps/server/api_server.go

```diff
@@ -2535,20 +2540,23 @@ func (a *apiServer) deletePipeline(ctx context.Context, request *pps.DeletePipel
 		return err
 	}
 
-	eg = errgroup.Group{}
-	// Delete cron input repos
-	if !request.KeepRepo {
-		pps.VisitInput(pipelineInfo.Details.Input, func(input *pps.Input) error {
-			if input.Cron != nil {
-				eg.Go(func() error {
-					return pachClient.DeleteRepo(input.Cron.Repo, request.Force)
-				})
-			}
-			return nil
-		})
-	}
-	if err := eg.Wait(); err != nil {
-		return err
+	// only do this if we were able to read the spec file
+	if pipelineInfo.Details != nil {
```

Comment:
I think having this makes logical sense. I'm only confused about the pickle the user may be in if we fail to get the pipeline details, but there are in fact cron repos to delete. I guess in that case they would need to manually debug anyway

Reply:
Yeah, I think if the spec really is gone for whatever reason we're stuck. It is the case that we create cron repos with a certain name pattern, but that doesn't seem safe enough. If I get my cron repos -> system repos change in this goes away, anyway